### PR TITLE
Add Wezterm support

### PR DESCRIPTION
**Pull request by Strosel**
Wezterm is a terminal emulator with built-in multiplexing and since I wanted to use Vim-slime with it, I went ahead and added that support. 

I actually did this a few months ago but forgot to make a PR, the plugin works as expected but its my first time messing about in vim-docs so I hope I didn't miss anything.

### CODE REVIEW

**Comment by jpalardy** [comment]
Hi @Strosel 

Thanks for the PR ⬆️ 

A while back, there was another PR (#333) that ended up not being merged. At the time, I saw the explosion of different targets I didn't support myself as a liability.

I ended up trying to extract "core vim-slime" from its support for targets -- https://github.com/jpalardy/vim-slime-ext-plugins

I know @Klafyvel ended up writing support for wezterm against it: https://github.com/Klafyvel/vim-slime-ext-wezterm

I'm not sure what @uncomfyhalomacro ended up doing 🤔 

…

That being said, and [vim-slime-ext-plugins](https://github.com/jpalardy/vim-slime-ext-plugins) being 90% done (but not DONE done), I'm willing to consider this PR.

Maybe you can integrate ideas from the other PRs (if needed), or coordinate with the people mentioned above. If it's working for all of you — I'll merge it in.

Let me know.

File: autoload/slime.vim [hunk 1]
```diff
@@ -25,7 +25,8 @@ function! s:ScreenSend(config, text)
         \ " -X eval \"readreg p " . g:slime_paste_file . "\"")
   call system("screen -S " . shellescape(a:config["sessionname"]) . " -p " . shellescape(a:config["windowname"]) .
         \ " -X paste p")
-  call system('screen -X colon ""')
+  call system('screen -X colon "
+"')
```
Comment (jpalardy) [inline autoload/slime.vim:29]: let's revert this line

this [happened](https://github.com/jpalardy/vim-slime/pull/333/files#diff-833e528646c9c9fd0ad2c302578b742106b4b7c5b87cc05c3df24babbb7db5ebR28) last time too 😄

File: doc/vim-slime.txt [hunk 2]
```diff
@@ -205,7 +206,22 @@ A pane relative to the currently active one in the target session
   - "up"/"down"/"right"/"left" for the pane in that direction relative to the location of the active pane
 
 ==============================================================================
-8. X11 Configuration 				*slime-x11*
+8. Wezterm Configuration 			*slime-wezterm*
+
+Wezterm is not the default, to use it you will have to add this line to your
+|.vimrc|:
+>
+	let g:slime_target = "kitty"
```
Comment (jpalardy) [inline doc/vim-slime.txt:214]:
```suggestion
	let g:slime_target = "wezterm"
```

copy-pasta?

**Comment by Klafyvel** [comment]
Hi! Yes, I use vim-slime-ext-plugins and the wezterm plugin I wrote on a daily basis. You can take some code from it to adapt here, like for choosing the target. You might be interested in bringing an improvement using

```bash
wezterm cli list-clients --format json
```

rather than the `wezterm cli list-clients` command I used.

File: doc/vim-slime.txt (hunk 2 above)
Comment (Strosel) [reply doc/vim-slime.txt:214]: Haha, yes i used kitty as a base but must have missed changing this

**Comment by jpalardy** [comment]
Hi @Strosel , @Klafyvel 

I think I'm ready to merge this in. Any more changes we want to make?

File: autoload/slime.vim [hunk 3]
```diff
@@ -100,6 +100,19 @@ function! s:ZellijConfig() abort
   else
     echoerr "Error: Allowed values are (current, right, left, up, down)"
   endif
```
Comment (albheim) [inline autoload/slime.vim:102]: Am I missing something or has the `endfunction` for `ZellijConfig` disappeared?

**Comment by Strosel** [comment]
No more changes on my end

**Comment by Klafyvel** [comment]
I haven't tested it as I run on vim-slime-ext, but I see nothing shocking here. :)

File: README.md [hunk 4]
```diff
@@ -339,6 +340,19 @@ vim.g.slime_get_jobid = function()
...
 
+### wezterm
+
+[wezterm](https://wezfurlong.org/wezterm/index.html) is *not* the default, to use it you will have to add this line to your .vimrc:
+
+    let g:slime_target = "wezterm"
+
+When you invoke vim-slime for the first time, you will be prompted for more configuration.
+
+weztem pane id
```
Comment (albheim) [inline README.md:351]:
```suggestion
wezterm pane id
```

File: README.md [hunk 5]
```diff
@@ -339,6 +340,19 @@ vim.g.slime_get_jobid = function()
...
+
+    This is the id of the kitty window that you wish to target.
```
Comment (albheim) [inline README.md:353]:
```suggestion
    This is the id of the wezterm pane that you wish to target.
```

**Comment by jpalardy** [comment]
Thanks @albheim ^

**Pull request merged**